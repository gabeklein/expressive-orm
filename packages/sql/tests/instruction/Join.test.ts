import { Join, Num, Query, Str, Type } from '../..';

class Foo extends Type {
  color = Str();

  bazValue = Join(where => {
    const bar = where(Bar, { color: this.color });
    const baz = where(Baz, { rating: bar.rating });

    return baz.value;
  })
}

class Bar extends Type {
  value = Str();
  color = Str();
  rating = Num();
}

class Baz extends Type {
  value = Str();
  rating = Num();
}

it.skip("will integrate query on select", () => {
  const query = new Query(where => {
    const foo = where(Foo);

    return {
      baz: foo.bazValue,
      color: foo.color
    }
  });

  expect(query).toMatchSnapshot();
})