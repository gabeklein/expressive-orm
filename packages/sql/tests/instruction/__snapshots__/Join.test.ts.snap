// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will integrate query on select 1`] = `
	select
	  \`Baz\`.\`value\` as \`baz\`,
	  \`Foo\`.\`color\` as \`color\`
	from
	  \`Foo\`
	  join \`Bar\` on \`Bar\`.\`color\` = \`Foo\`.\`color\`
	  join \`Baz\` on \`Baz\`.\`rating\` = \`Bar\`.\`rating\`
`;
